opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+/**
+ * Tracer is the entry point API between instrumentation code and the tracing implementation.
+ */
+public interface Tracer {
+
+  /**
+   * Starts a new trace and creates a new root span.
+   *
+   * <p>This method should be used by services that are instrumented for tracing but did not receive
+   * trace ID from upstream request.
+   *
+   * @param operationName the service's own name for the end-point that received the request
+   * represented by this trace and span. The domain of names must be limited, e.g. do not use UUIDs
+   * or entity IDs or timestamps as part of the name.
+   */
+  Span startTrace(String operationName);
+
+  /**
+   * Joins a trace started elsewhere and creates a new span as a child of the given parent trace
+   * context.
+   *
+   * <p>This method should be used by services that receive tracing info from upstream.
+   *
+   * @param operationName the service's own name for the end-point that received the request
+   * represented by this trace and span. The domain of names must be limited, e.g. do not use UUIDs
+   * or entity IDs or timestamps as part of the name.
+   * @param parent Trace Context of a client span started elsewhere and whose trace we're joining.
+   */
+  Span joinTrace(String operationName, TraceContext parent);
```

Comment:
so the Go API has `startTrace` and `joinTrace` equivs, but also this:

https://github.com/opentracing/opentracing-go/blob/master/tracer.go#L46

Python has something similar in that a `Span` can be constructed given an explicit `TraceContext` to bind to.

@yurishkuro: did we add that so that the golang Zipkin impl could retain the same `span_id` on both sides of an RPC? Or was it just to be more general? In any case, we should remove it in Go/Python xor add it here.